cmd/PersistFFISData/main.go

```diff
@@ -0,0 +1,63 @@
+// Package main compiles to an AWS Lambda handler binary that, when invoked,
+// parses the JSON found in the event payload for FFIS data, and upserts it
+// into found grants records.
+
+package main
+
+import (
+	"context"
+	"fmt"
+	goLog "log"
+
+	ddlambda "github.com/DataDog/datadog-lambda-go"
+	goenv "github.com/Netflix/go-env"
+	"github.com/aws/aws-lambda-go/events"
+	"github.com/aws/aws-lambda-go/lambda"
+	"github.com/aws/aws-sdk-go-v2/service/dynamodb"
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+	"github.com/usdigitalresponse/grants-ingest/internal/awsHelpers"
+	"github.com/usdigitalresponse/grants-ingest/internal/log"
+	awstrace "gopkg.in/DataDog/dd-trace-go.v1/contrib/aws/aws-sdk-go-v2/aws"
+)
+
+type Environment struct {
+	LogLevel          string `env:"LOG_LEVEL,default=INFO"`
+	DestinationTable  string `env:"GRANTS_PREPARED_DYNAMODB_NAME,required=true"`
+	UsePathStyleS3Opt bool   `env:"S3_USE_PATH_STYLE,default=false"`
+	Extras            goenv.EnvSet
+}
+
+var (
+	env    Environment
+	logger log.Logger
+	// sendMetric       = ddHelpers.NewMetricSender("PersistFFISData", "source:grants.gov")
```

Comment:
I've been copying these in from the original pattern/template - can I remove?

Reply:
Normally, yes – no expectation to keep unused code/comments for the sake of convention – but in this case, I think it would be good to emit an `opportunity.saved` metric after successfully saving to DDB. I dropped a suggestion in the `handleS3Event()` function accordingly, although it could also be sent from `UpdateOpportunity()` if you think that's more appropriate. The idea is to have this mirror the `opportunity.saved` metric [emitted by `PersistGrantsGovXMLDB`](https://github.com/usdigitalresponse/grants-ingest/blob/main/cmd/PersistGrantsGovXMLDB/handler.go#LL96C2-L96C36), allowing us to track DB updates across both sources.

Following that, this should be uncommented and the `source` tag should be updated to reflect FFIS rather than Grants.gov:
```suggestion
	sendMetric = ddHelpers.NewMetricSender("PersistFFISData", "source:ffis")
```